### PR TITLE
`Development`: Add gitHub action outputting the changed files

### PR DESCRIPTION
**Pull request by Jan-Thurner**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

Currently, there is no feasible way to compare and analyze endpoint connections. This has to be done manually. As there are many endpoints and the generation of URIs is often profoundly nested, this is tedious and error-prone. Developers, maintainers, and reviewers have to spend significant amounts of time going through the process over and over again for every PR. 

As for all involved parties, only the changes of the specific PR are relevant, they should be the only ones shown to them. This makes the review process less overwhelming and faster.

### Description

This GitHub action forms the foundation of the analysis of endpoint connections on Artemis. It will print the files changed in a PR to the GitHub action's output. As for a single PR, only the changes made within that PR are to be code-reviewed by developers and maintainers, the analysis of endpoint connections focuses only on modified parts of the code, ensuring the feedback is relevant and concise.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

## Summary by CodeRabbit

- **New Features**
	- Enhanced GitHub Actions workflow to analyze and display modified files in specific directories on pull requests.

### CODE REVIEW

File: .github/workflows/alaysis-of-endpoint-connections.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      - '.github/**'
+      - '!.github/workflows/build.yml'
+      - '!.github/workflows/testserver.yml'
+
+jobs:
+  show-modified-files:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (Strohgelaender) [inline .github/workflows/alaysis-of-endpoint-connections.yml:18]:
```suggestion
        uses: actions/checkout@v4
```

File: .github/workflows/alaysis-of-endpoint-connections.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  show-modified-files:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Get list of modified files
+        id: files
+        run: |
+          git diff --name-only ${{ github.event.pull_request.base.ref }}..${{ github.event.pull_request.head.ref }} > modified_files.txt
```
Comment (Hialus) [inline .github/workflows/alaysis-of-endpoint-connections.yml:23]:
```suggestion
          git diff --name-only ${{ github.event.pull_request.base.ref }} HEAD > modified_files.txt
```

File: .github/workflows/analysis-of-endpoint-connections.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      - 'LICENSE'
+      - 'SECURITY.md'
+      - 'docs/**'
+
+jobs:
+  show-modified-files:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
```
Comment (Hialus) [inline .github/workflows/analysis-of-endpoint-connections.yml:18]:
```suggestion
        uses: actions/checkout@v4
        with:
          fetch-depth: 0
```

File: .github/workflows/analysis-of-endpoint-connections.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      - 'src/main/webapp/**'
+
+jobs:
+  show-modified-files:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```
Comment (Hialus) [inline .github/workflows/analysis-of-endpoint-connections.yml:16]: Optional: You could consider looking into a solution for this that is fast.

**Review by Hialus — DISMISSED** [review]
Looked through the action with Jan in person and it looks good 🚀 

Just one thing: It would be great if you would describe in your PR description why this is needed for your following work.

**Review by pzdr7 — APPROVED** [review]
LGTM

**Review by MarkusPaulsen — COMMENTED** [review]
Left some comments.

File: .github/workflows/analysis-of-endpoint-connections.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+      - name: Get list of modified files
+        id: files
```
[the remark this answers is not in the document]
Comment (Jan-Thurner) [reply .github/workflows/analysis-of-endpoint-connections.yml:19]: The id is needed to access the step's output at a later stage. It is groundwork for future additions.

File: .github/workflows/analysis-of-endpoint-connections.yml (hunk 4 above)
Comment (Jan-Thurner) [reply .github/workflows/analysis-of-endpoint-connections.yml:25]: Will do in a followup

**Review by pzdr7 — APPROVED** [review]

**Review by MaximilianAnzinger — APPROVED** [review]

**Review by florian-glombik — APPROVED** [review]
Looks good to me _(as someone with basic knowledge about GitHub actions)_

**Pull request merged**